### PR TITLE
Some fixes for bringing up GCE E2E cluster

### PR DESCRIPTION
**Pull request by pmorie**
Some fixes found while bringing up a GCE cluster from fedora.

### CODE REVIEW

**Comment by pmorie** [comment]
@kubernetes/rh-cluster-infra PTAL

File: pkg/kubectl/cmd/config/config.go [hunk 1]
```diff
@@ -225,6 +225,9 @@ func ModifyConfig(configAccess ConfigAccess, newConfig clientcmdapi.Config, rela
 
 			configToWrite := getConfigFromFileOrDie(destinationFile)
 			t := *cluster
+			if configToWrite.Clusters == nil {
```
Comment (ncdc) [inline pkg/kubectl/cmd/config/config.go]: Do we want/need a unit test for this?

Comment (ashcrow) [inline pkg/kubectl/cmd/config/config.go]: :+1: I'm happy to do a follow on PR adding unittests for this if that works for @pmorie

**Comment by pmorie** [comment]
@kubernetes/goog-testing

File: pkg/kubectl/cmd/config/config.go (hunk 1 above)
Comment (pmorie) [inline pkg/kubectl/cmd/config/config.go]: This code probably should move into `getConfigFromFileOrDie`

**Comment by ncdc** [comment]
@liggitt @deads2k for the config.go change

File: pkg/kubectl/cmd/config/config.go (hunk 1 above)
Comment (deads2k) [inline pkg/kubectl/cmd/config/config.go]: > This code probably should move into getConfigFromFileOrDie

`NewConfig` already tries: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/api/types.go#L108 .  Its call in the `Load` path and the `getConfigFromFileOrDie` path.  Is the `Decode` (https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/clientcmd/loader.go#L255) setting them back to nil?  I don't see much value in getting a nil map versus an empty map, so I'm in favor of fixing `Load`

**Comment by pmorie** [comment]
@ixdy @mikedanese

File: pkg/kubectl/cmd/config/config.go [hunk 2]
```diff
@@ -429,6 +430,10 @@ func getConfigFromFileOrDie(filename string) *clientcmdapi.Config {
 		return clientcmdapi.NewConfig()
 	}
 
+	if config.Clusters == nil {
+		config.Clusters = map[string]*clientcmdapi.Cluster{}
```
Comment (ixdy) [inline pkg/kubectl/cmd/config/config.go]: `clientcmdapi.NewConfig()` appears to be setting `config.Clusters` to a non-nil value; I wonder if `clientcmdlatest.Codec.DecodeInto()` is unsetting it?

Comment (pmorie) [inline pkg/kubectl/cmd/config/config.go]: I believe that the codec overrides whatever values are in the object.  This is what I would expect from an API standpoint -- if I point the codec at an empty file, which is valid yaml, i'd expect to get a zeroed object back.

**Comment by pmorie** [comment]
@ixdy PTAL, I'd like to get this merged as I have another PR in-progress that depends on it.

File: pkg/kubectl/cmd/config/config.go (hunk 2 above)
Comment (liggitt) [inline pkg/kubectl/cmd/config/config.go]: @smarterclayton, you just did some work in this area... does this look like what you would expect here?

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -740,7 +740,7 @@ function kube-up {
 
   # curl in mavericks is borked.
   secure=""
-  if which sw_vers > /dev/null; then
+  if which sw_vers >& /dev/null; then
```
Comment (ixdy) [inline cluster/gce/util.sh]: did you mean` &>`?

File: pkg/kubectl/cmd/config/config.go (hunk 2 above)
Comment (ixdy) [inline pkg/kubectl/cmd/config/config.go]: I'm wondering if maybe this should be fixed in `LoadFromFile` instead?

also, there are several maps in `Config` which could presumably be set `nil` by `LoadFromFile`. Should we cover all of them? should we expect callers to check for nil-ness?

Comment (liggitt) [inline pkg/kubectl/cmd/config/config.go]: @deads2k since you just looked at @smarterclayton's change

Comment (deads2k) [inline pkg/kubectl/cmd/config/config.go]: > I'm wondering if maybe this should be fixed in LoadFromFile instead?
> 
> also, there are several maps in Config which could presumably be set nil by LoadFromFile. Should we cover all of them? should we expect callers to check for nil-ness?

For this structure, I don't see any value in making callers distinguish between nil and empty.  I'd make this change in `LoadFromFile` for all maps there.

**Comment by ixdy** [comment]
um, why did this get merged? there was no LGTM label on this.

@eparis @brendandburns

**Comment by ixdy** [comment]
oh, because #16986 was merged.

**Comment by ixdy** [comment]
please address comments and reopen/create a new PR with changes.

**Comment by pmorie** [comment]
@ixdy My bad, I had momentarily forgotten about this PR when I asked vish to take a look at the other one.  Oops! :sob:

**Pull request merged**